frontend/src/components/_shared/UserAvatar.tsx

```diff
@@ -0,0 +1,40 @@
+import { Avatar, AvatarFallback, AvatarImage } from "@components/ui/avatar";
+import { cn } from "@lib/utils";
+import { User } from "@portaljs/ckan";
+import { getUser } from "@utils/user";
+import { useSession } from "next-auth/react";
+import { useEffect, useState } from "react";
+
+export default function UserAvatar({
+  id,
+  className,
+}: {
+  id: string;
+  className?: string;
+}) {
+  const { data: sessionData } = useSession();
+  const [user, setUser] = useState<User>();
+  const apiKey = sessionData?.user.apikey ?? "";
+
+  useEffect(() => {
+    getUser({ apiKey, id }).then((v) => setUser(v as User));
```

Comment:
I think it would be better if you use a tRPC route instead of calling the getUset function in an useEffect.